groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/completion/SpockUtils.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.groovy.editor.completion;
+
+
+import java.util.HashSet;
+import java.util.Set;
+import org.codehaus.groovy.ast.ClassNode;
+import org.netbeans.api.lexer.Token;
+import org.netbeans.api.lexer.TokenSequence;
+import org.netbeans.modules.groovy.editor.api.completion.util.CompletionContext;
+import org.netbeans.modules.groovy.editor.api.lexer.GroovyTokenId;
+import org.netbeans.modules.groovy.editor.api.lexer.LexUtilities;
+
+/**
+ *
+ * @author Petr Pisl
+ */
+public class SpockUtils {
+    
+    public static boolean isInSpecificationClass(CompletionContext context) {
+        
+        Set<String> visited = new HashSet<>();
+        String name;
+        ClassNode classNode = context.declaringClass;
```

Comment:
Idea: If there was a `GroovyParserResut` available from the `CompletionContext`, one could do 
```
context.declaringClass.isDerivedFrom(
   parserResult.resolveClassName("spock.lang.Specification")
)
```

Reply:
Good idea.

Reply:
Done